### PR TITLE
Add cert-rotation configmap for setting base rotation interval

### PR DESCRIPTION
**Pull request by soltysh**
/assign @mfojtik @sttts

### CODE REVIEW

File: pkg/cmd/render/render.go [hunk 1]
```diff
@@ -73,6 +79,9 @@ func (r *renderOpts) AddFlags(fs *pflag.FlagSet) {
 	fs.StringArrayVar(&r.etcdServerURLs, "manifest-etcd-server-urls", r.etcdServerURLs, "The etcd server URL, comma separated.")
 	fs.StringVar(&r.etcdServingCA, "manifest-etcd-serving-ca", r.etcdServingCA, "The etcd serving CA.")
 	fs.StringVar(&r.clusterConfigFile, "cluster-config-file", r.clusterConfigFile, "Openshift Cluster API Config file.")
+
+	fs.StringVar(&r.certRotationBase, "unsupported-cert-rotation-base", r.certRotationBase, "UNSUPPORTED! Certificate rotation base unit.")
```
Comment (sttts) [inline pkg/cmd/render/render.go]: isn't there a time.Duration variant for that?

File: pkg/cmd/render/render.go [hunk 2]
```diff
@@ -199,3 +214,20 @@ func discoverCIDRs(clusterConfigFileData []byte, renderConfig *TemplateData) err
...
+
+func writeCertRotationConfig(assetOutputDir, certRotationBase string) error {
+	f := filepath.Join(assetOutputDir, "manifests", "0000_10_kube-apiserver-operator_03_certrotation.yaml")
+	if err := os.MkdirAll(filepath.Dir(f), os.FileMode(0755)); err != nil {
+		return err
+	}
+	certRotationConfig := corev1.ConfigMap{
+		TypeMeta:   metav1.TypeMeta{APIVersion: corev1.SchemeGroupVersion.String(), Kind: "ConfigMap"},
+		ObjectMeta: metav1.ObjectMeta{Name: "unsupported-cert-rotation-config", Namespace: operatorclient.OperatorNamespace},
+		Data:       map[string]string{"base": certRotationBase},
```
Comment (sttts) [inline pkg/cmd/render/render.go]: ic, you just pass through the string?

File: pkg/operator/certrotationcontroller/certrotationcontroller.go [hunk 3]
```diff
@@ -65,13 +65,20 @@ func NewCertRotationController(
...
+	}
+
 	certRotator, err := certrotation.NewCertRotationController(
 		"AggregatorProxyClientCert",
 		certrotation.SigningRotation{
 			Namespace:     operatorclient.OperatorNamespace,
 			Name:          "aggregator-client-signer",
-			Validity:      30 * rotationDay,
-			Refresh:       15 * rotationDay,
+			Validity:      30 * rotationBase,
```
Comment (sttts) [inline pkg/operator/certrotationcontroller/certrotationcontroller.go]: would leave the name rotationDay. Makes it easier to read.

File: pkg/operator/starter.go [hunk 4]
```diff
@@ -142,12 +142,19 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
 		ctx.EventRecorder,
 	)
 
+	certRotationBase := time.Duration(0)
+	if certRotationConfig, err := kubeClient.CoreV1().ConfigMaps(operatorclient.OperatorNamespace).Get("unsupported-cert-rotation-config", metav1.GetOptions{}); err == nil {
```
Comment (sttts) [inline pkg/operator/starter.go]: better just ignore NotFound errors, but retry on different errors.

Comment (sttts) [reply pkg/operator/starter.go]: and fail if retries fail as well.

File: pkg/cmd/render/render.go (hunk 1 above)
Comment (soltysh) [reply pkg/cmd/render/render.go]: Good point, it's worth this being `time.Duration`.

File: pkg/operator/starter.go (hunk 4 above)
Comment (soltysh) [reply pkg/operator/starter.go]: addressed

File: pkg/operator/certrotationcontroller/certrotationcontroller.go [hunk 5]
```diff
@@ -44,7 +44,7 @@ func NewCertRotationController(
 	operatorClient v1helpers.StaticPodOperatorClient,
 	configInformer configinformers.SharedInformerFactory,
 	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
-	eventRecorder events.Recorder,
+	eventRecorder events.Recorder, certRotationBase time.Duration,
```
Comment (sttts) [inline pkg/operator/certrotationcontroller/certrotationcontroller.go]: also here `certRotationDay`

Comment (sttts) [reply pkg/operator/certrotationcontroller/certrotationcontroller.go]: or even just `day`

File: pkg/operator/starter.go [hunk 6]
```diff
@@ -142,12 +143,28 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
 		ctx.EventRecorder,
 	)
 
+	certRotationBase := time.Duration(0)
+	err = wait.PollImmediate(time.Second, 1*time.Minute, func() (bool, error) {
+		certRotationConfig, err := kubeClient.CoreV1().ConfigMaps(operatorclient.OperatorNamespace).Get("unsupported-cert-rotation-config", metav1.GetOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			return false, err
+		}
+		if value, ok := certRotationConfig.Data["base"]; ok {
```
Comment (sttts) [inline pkg/operator/starter.go]: the NotFound case is missing, isn't it? We have to exit the loop

**Pull request merged**